components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQConnectionConsumer.java

```diff
@@ -149,66 +149,54 @@ private void startConsumer() throws ShutdownSignalException, IOException {
         }
 
         while (isActive()) {
+            RabbitMQMessage message;
             try {
                 if (!channel.isOpen()) {
                     channel = queueingConsumer.getChannel();
                 }
-                channel.txSelect();
-            } catch (IOException e) {
-                log.error("Error while starting transaction", e);
-                continue;
-            }
-
-            boolean successful = false;
-            boolean mediationError = false;
-
-            RabbitMQMessage message = null;
-            try {
                 message = getConsumerDelivery(queueingConsumer);
-            } catch (InterruptedException e) {
-                log.error("Error while consuming message", e);
+            } catch (InterruptedException | ShutdownSignalException | ConsumerCancelledException e) {
+                log.warn("Exception occurred while consuming the message", e);
                 continue;
             }
 
             if (message != null) {
                 idle = false;
+                RabbitMQAckStates ackState = RabbitMQAckStates.REJECT_AND_REQUEUE;
+                boolean mediationError = false;
                 try {
-                    successful = injectHandler.invoke(message, inboundName);
+                    ackState = injectHandler.invokeAndReturnAckState(message, inboundName);
                 } catch (Exception e) {         //we need to handle any exception upon injecting to mediation
-                    successful = false;
+                    ackState = RabbitMQAckStates.REJECT_AND_REQUEUE;
```

Comment:
We need to evaluate the value of the property, 'SET_REQUEUE_ON_ROLLBACK'. WDYT?

Reply:
Done in db57000c3482bf048fdfba088070458e5de91930